.github/workflows/build.yaml

```diff
@@ -63,12 +63,20 @@ jobs:
     steps:
       - name: git clone
         uses: actions/checkout@v2
+
       - name: tool/bots.sh
         env:
           BOT: ${{ matrix.bot }}
           PLATFORM: vm
         run: ./tool/bots.sh
 
+      - name: image failures
+        uses: actions/upload-artifact@v1
+        if: failure() # Only if failure then failures directory exists.
+        with:
+          name: test-image-failures-${{ matrix.os }} # Name for the artifact
```

Comment:
The "matrix.os" here is currently being replaced with an empty string (as there's no `os` in the `matrix` config). It would be removed, although if it may be run on multiple OSes in the future, perhaps it's worth leaving.